### PR TITLE
Jenkins copyright checks

### PR DESCRIPTION
**Pull request by AdamBrousseau**
Initial commit of copyrightCheck file
    
- Get list of added, copied, modified files
- For each file ensure there is a valid copyright line
- Ensure the second year matches current year
- Collect files that do not pass check and print at end of run
- Fail if at least 1 file fails check

[skip ci]

Fixes Issue #64

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
Jenkins copyright check

**Comment by AdamBrousseau** [comment]
Example of failed case
https://ci.eclipse.org/openj9/job/PullRequest-CopyrightCheck/54/console
https://github.com/AdamBrousseau/openj9/pull/8

@keithc-ca for review

**Comment by AdamBrousseau** [comment]
(unlrelated)
jenkins test sanity zlinux

**Comment by AdamBrousseau** [comment]
jenkins test sanity zlinux

File: buildenv/jenkins/copyrightCheck [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            echo FILES
+            def FILES_LIST = FILES.split("\\r?\\n")
+            DATE_YEAR = sh (
+                script: "date +%Y",
+                returnStdout: true
+            ).trim()
+            FILES_LIST.each() {
+                println "Checking file: '${it}'"
+                RESULT = sh (
+                    script: "grep -qE \"Copyright \\(c\\) ([0-9]{4}), ${DATE_YEAR} IBM Corp. and others\" '${it}'",
```
Comment (keithc-ca) [inline buildenv/jenkins/copyrightCheck]: If you use single quotes around the pattern, I expect fewer escapes would be required.

Comment (AdamBrousseau) [reply buildenv/jenkins/copyrightCheck]: updated commit, seems to still work properly

File: buildenv/jenkins/copyrightCheck [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            if (FAIL) {
+                echo "${HASHES}"
+                echo "The following files were modified and have incorrect Copyrights"
+                BAD_FILES.each() {
+                    echo "${it}"
+                }
+                echo "${HASHES}"
+                sh 'exit 1'
+            } else {
+                echo "All modified files appear to have correct Coprights"
```
Comment (keithc-ca) [inline buildenv/jenkins/copyrightCheck]: That's not how you spell 'copyrights'. ;-)

File: buildenv/jenkins/copyrightCheck (hunk 1 above)
Comment (keithc-ca) [reply buildenv/jenkins/copyrightCheck]: I suspect you didn't have `\` escapes before, but it seems right now.

File: buildenv/jenkins/copyrightCheck (hunk 2 above)
Comment (AdamBrousseau) [reply buildenv/jenkins/copyrightCheck]: fixed

**Pull request merged**